### DISCUSSION TITLE
Building the project on ubuntu 2018

### DISCUSSION THREAD

**Discussion by pierreloicq** [osmandapp/OsmAnd#12146] [General]
Hi,
I am trying to build the project on ubuntu 2018 based on [this page](https://github.com/osmandapp/OsmAnd-core)

So far, I changed/added this : 
> sudo apt-get install openjdk-8-jdk (version 7 is no more found)
> sudo apt-get install gcc gcc-multilib g++ g++-multilib (without version number, 4.9 is no more found)

So I removed the numbers acordingly in build/targets/amd64-linux-gcc.sh

> sudo apt-get install swig
> sudo apt install p7zip-full

But then with ./build/amd64-linux-gcc.sh, I manage to go until : 

> -- Found SWIG: /usr/bin/swig3.0 (found version "3.0.12") 
> -- Found Java: /usr/bin/java (found version "1.8.0.292") found components:  Development 
> -- Found JNI: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/libjawt.so  
> -- Found Java: /usr/bin/java (found version "1.8.0.292") 
> [....]
> -- Configuring done
> CMake Error at /home/pierre/projetCO2/core/samples/java/CMakeLists.txt:21 (add_dependencies):
>   The dependency target "OsmAndCore_ResourcesBundle_shared" of target
>   "OsmAndCore_JavaSample1" does not exist.

What do I miss? Are openjdk v7 and gcc v4.9 mandatory ?

Thanks

**Comment by vshcherb** [comment]
This note might be outdated cause we switched to clang. Also we don't build ubuntu version anymore and use only ios / android. 
The only library we build for MapCreator is using these instructions
```
  (cd "resources" && ./verify-embed-resources-list.sh)
   alias clang='clang -std=c++11'
    "build/amd64-linux-clang.sh" release

    (cd "baked/amd64-linux-clang-release.make" && make -j4)
```

**Reply by pierreloicq** [reply]
Thank you. Is everything in this note outdated ? Is so, where can I find the building process ?

**Reply by vshcherb** [reply]
In my response how we build it for OsmAndMapCreator for Ubuntu. For xcode https://docs.osmand.net/en/main@latest/development/build-osmand/how-to-compile-the-ios-version

Updated as well here
https://docs.osmand.net/en/main@latest/development/build-osmand/how-to-compile-mapcreator-and-tools#compile-qt-core-version-for-osmandmapcreator

**Reply by pierreloicq** [reply]
Thanks.
On [this page](https://docs.osmand.net/en/main@latest/development/build-osmand/how-to-compile-the-android-version) it is not clear if I need to install android studio first or if the lines like 
`export ANDROID_SDK=/[your-path]/Android/sdk`
are path for future installation ?

**Reply by pierreloicq** [reply]
Ok I installed android studio.
Building fails on the tests (log attached)
Is it expected ?
Thanks
[build_stop_on_tests.pdf](https://github.com/osmandapp/OsmAnd/files/6836863/build_stop_on_tests.pdf)
